### PR TITLE
CON-2598 Add who will enter details page

### PR DESCRIPTION
**Pull request by ben1stone**

### CODE REVIEW

File: src/SFA.DAS.EmployerCommitmentsV2.Web.UnitTests/Validators/WhoWillEnterTheDetailsViewModelValidatorTests.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        public void Arrange()
+        {
+            _autoFixture = new Fixture();
+            _validator = new WhoWillEnterTheDetailsViewModelValidator();
+        }
+
+        [Test]
+        public void WhenValidatingWhoWillEnterTheDetails_AndSelectionIsNotMade_ThenValidatorReturnsInvalid()
+        {
+            var viewModel = _autoFixture.Build<WhoWillEnterTheDetailsViewModel>().Create();
```
Comment (cofaulco) [inline src/SFA.DAS.EmployerCommitmentsV2.Web.UnitTests/Validators/WhoWillEnterTheDetailsViewModelValidatorTests.cs:27]:
```suggestion
            var viewModel = _autoFixture.Create<WhoWillEnterTheDetailsViewModel>();
```

Comment (cofaulco) [reply src/SFA.DAS.EmployerCommitmentsV2.Web.UnitTests/Validators/WhoWillEnterTheDetailsViewModelValidatorTests.cs:27]: Can be simplified if not making any modifications on Build();

**Review by cofaulco — COMMENTED** [review]
Are we feature toggling this new journey so that we can continue to develop and deploy with placeholder views?

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Validators/WhoWillEnterTheDetailsViewModelValidator.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿using FluentValidation;
+using SFA.DAS.EmployerCommitmentsV2.Web.Models.Apprentice;
+
+namespace SFA.DAS.EmployerCommitmentsV2.Web.Validators
+{
+    public class WhoWillEnterTheDetailsViewModelValidator : AbstractValidator<WhoWillEnterTheDetailsViewModel>
+    {
+        public WhoWillEnterTheDetailsViewModelValidator()
+        {
+            RuleFor(r => r.AccountHashedId).NotEmpty();
```
Comment (cofaulco) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Validators/WhoWillEnterTheDetailsViewModelValidator.cs:10]: What error message is returned to the user for these top 3 rules? Does it make sense to the user as these are not something they have submitted?

Comment (ben1stone) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Validators/WhoWillEnterTheDetailsViewModelValidator.cs:10]: Removed as discussed

File: src/SFA.DAS.EmployerCommitmentsV2.Web.UnitTests/Controllers/ApprenticeControllerTests/WhenPostingEnterNewTrainingProvider.cs [hunk 3]
```diff
@@ -27,18 +26,27 @@ public void Arrange()
...
-            var result = await _fixture.EnterNewTrainingProvider();
+            var result = await _fixture.EnterNewTrainingProvider(true);
 
-            _fixture.VerifyRedirectsToSendNewTrainingProviderRequest(result);
-        } 
+            _fixture.VerifyRedirectsToWhoWillEnterTheDetailsPage(result);
+        }
+
+        [Test]
+        public async Task AndTheFeatureToggleIsDisabled_ThenRedirectToWhoWillEnterTheDetailsPage()
```
Comment (cofaulco) [inline src/SFA.DAS.EmployerCommitmentsV2.Web.UnitTests/Controllers/ApprenticeControllerTests/WhenPostingEnterNewTrainingProvider.cs:37]: name of this test should be `ThenRedirectToSendRequestPage`

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Mappers/Apprentice/EnterNewTrainingProviderViewModelToSendNewTrainingProviderRequestMapper.cs [hunk 4]
```diff
@@ -1,10 +1,10 @@
-﻿using System.Threading.Tasks;
-using SFA.DAS.CommitmentsV2.Shared.Interfaces;
+﻿using SFA.DAS.CommitmentsV2.Shared.Interfaces;
 using SFA.DAS.EmployerCommitmentsV2.Web.Models.Apprentice;
+using System.Threading.Tasks;
 
 namespace SFA.DAS.EmployerCommitmentsV2.Web.Mappers.Apprentice
 {
-    public class SendNewTrainingProviderRequestMapper : IMapper<EnterNewTrainingProviderViewModel, SendNewTrainingProviderRequest>
+    public class EnterNewTrainingProviderViewModelToSendNewTrainingProviderRequestMapper : IMapper<EnterNewTrainingProviderViewModel, SendNewTrainingProviderRequest>
```
Comment (cofaulco) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Mappers/Apprentice/EnterNewTrainingProviderViewModelToSendNewTrainingProviderRequestMapper.cs:7]: This is a mouthful 😅

**Review by cofaulco — APPROVED** [review]

**Pull request merged**